src/nupic/encoders/CoordinateEncoder.cpp

```diff
@@ -0,0 +1,128 @@
+/* ---------------------------------------------------------------------
+ * Numenta Platform for Intelligent Computing (NuPIC)
+ * Copyright (C) 2019, David McDougall
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses.
+ *
+ * http://numenta.org/licenses/
+ * --------------------------------------------------------------------- */
+
+/** @file
+ * Implement the CoordinateEncoder
+ */
+
+#define _USE_MATH_DEFINES // Define before including anything
+#include <nupic/encoders/CoordinateEncoder.hpp>
+#include <math.h> // tgamma, M_PI
+#include <nupic/utils/MurmurHash3.hpp>
+
+using namespace std;
+using namespace nupic::encoders;
+
+CoordinateEncoder::CoordinateEncoder(const CoordinateEncoderParameters &parameters)
+  { initialize( parameters ); }
+
+void CoordinateEncoder::initialize( const CoordinateEncoderParameters &parameters ) {
+  // Check parameters:  This has all of the same constraints as the RDSE does.
+  // The RDSE will raise an exception if the parameters are bad.
+  RDSE rdse( parameters );
```

Comment:
the exception will be confusing, RDSE failed when you don't use RDSE? (only Coord); 
Do some try-catch, print the exception, and add "Coord failed: its params have the same restrictions as RDSE"